core/modules/comment/comment.module

```diff
@@ -1205,6 +1207,25 @@ function comment_form_node_type_form_alter(&$form, $form_state) {
         BACKDROP_REQUIRED => t('Required'),
       ),
     );
+    $form['comment']['comment_close_enable'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Automatically close comments on old content'),
+      '#default_value' => $node_type->settings['comment_close_enable'],
+    );
+    $form['comment']['comment_close_days'] = array(
+      '#type' => 'textfield',
+      '#size' => 4,
+      '#maxlength' => 4,
+      '#states' => array(
+        'invisible' => array(
+          ':input[name="comment_close_enable"]' => array('checked' => FALSE),
+        ),
+      ),
+      '#field_prefix' => t('Close comments after'),
+      '#default_value' => $node_type->settings['comment_close_days'],
+      '#field_suffix' => t('days'),
+      '#element_validate' => array('_comment_close_days_validate'),
```

Comment:
I think we could remove this special validation function by using #type = 'number' and then setting #min and #step.